pkmodel/solution.py

```diff
@@ -12,6 +18,182 @@ class Solution:
         an example paramter
 
     """
-    def __init__(self, value=44):
-        self.value = value
+    def __init__(self, dose_dataframe,model_instance):
+        self.dose_dataframe = dose_dataframe
+        self.model_instance = model_instance
 
+    def dose(self,t,row_index):
```

Comment:
Explain what this function does.
""" This function sets the doses at a certain time point to the value given in the dosing protocol. The dose is set to 0 in case of an instantaneous dosing because this dose is taken into account when setting the initial values. 

:param: ....

"""